src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Engine.java

```diff
@@ -472,27 +473,54 @@ protected MappedEntity createPreparedStatementForInserts(final DbEntity entity)
         logger.trace(statement);
 
 
-        final PreparedStatement ps, psReturn, psWithAutoInc;
+        final PreparedStatement ps, psReturn, psWithAutoInc, psMerge;
         try {
 
             // Generate keys when the table has at least 1 column with auto generate value.
             final int generateKeys = columnWithAutoIncName != null? Statement.RETURN_GENERATED_KEYS : Statement.NO_GENERATED_KEYS;
             ps = this.conn.prepareStatement(statement, generateKeys);
             psReturn = this.conn.prepareStatement(insertReturnStatement, generateKeys);
             psWithAutoInc = this.conn.prepareStatement(statementWithAutoInt, generateKeys);
-
+            psMerge = this.conn.prepareStatement(statementWithMerge, generateKeys);
             return new MappedEntity()
                 .setInsert(ps)
                 .setInsertReturning(psReturn)
                 .setInsertWithAutoInc(psWithAutoInc)
                 // The auto incremented column must be set, so when persisting a row, it's possible to retrieve its value
                 // by consulting the column name from this MappedEntity.
-                .setAutoIncColumn(columnWithAutoIncName);
+                .setAutoIncColumn(columnWithAutoIncName)
+                .setUpsert(psMerge);
```

Comment:
See, if we rename the below method, we'll be avoiding this misalignment in the naming convention (ie. `.setUpsert(psUpsert)`).